### PR TITLE
 Fix the fetching of LDAP users in the LocalAddressbook 

### PR DESCRIPTION
**Pull request by LEDfan**
This PR replaces #12784. AFAIK this should be merged in 8.1.

### CODE REVIEW

**Comment by LEDfan** [comment]
Tested with LDAP backend.

File: lib/private/contacts/localaddressbook.php [hunk 1]
```diff
@@ -56,10 +56,16 @@ public function search($pattern, $searchProperties, $options) {
 				if($property === 'FN') {
 					$result = $this->userManager->searchDisplayName($pattern);
 				} else if ($property === 'id') {
-					$result = $this->userManager->search($pattern);
+					if(in_array('get', $options)){
```
Comment (DeepDiver1975) [inline lib/private/contacts/localaddressbook.php]: $options should be key-value pairs - furthermore this is not really intuitive

Comment (LEDfan) [inline lib/private/contacts/localaddressbook.php]: I did it with the $options, because I didn't know a better option. Maybe you do?

Comment (LEDfan) [inline lib/private/contacts/localaddressbook.php]: An extra function getContact($id) would also solve this. IMO that's better.

**Comment by LEDfan** [comment]
@DeepDiver1975 Should I change something before this can be merged? Tnx!

**Comment by LukasReschke** [comment]
@owncloud-bot Retest this please.

**Comment by lunarok** [comment]
Hi, this PR make it possible for LDAP user to use the Chat application.

**Comment by MorrisJobke** [comment]
@owncloud-bot Retest this please

**Comment by MorrisJobke** [comment]
@LEDfan A unit test would be nice

**Comment by MorrisJobke** [comment]
Maybe we missed something, therefore maybe @blizzz can also have a short look.

**Comment by MorrisJobke** [comment]
```
PHP Fatal error:  Call to a member function getUID() on null in /var/jenkins/workspace/pull-request-analyser-ng-simple@3/label/SLAVE/lib/private/contacts/localaddressbook.php on line 76
00:02:56.738 PHP Stack trace:
00:02:56.738 PHP   1. {main}() /usr/local/bin/phpunit:0
00:02:56.738 PHP   2. PHPUnit_TextUI_Command::main() /usr/local/bin/phpunit:612
00:02:56.745 PHP   3. PHPUnit_TextUI_Command->run() phar:///usr/local/bin/phpunit/phpunit/TextUI/Command.php:138
00:02:56.745 PHP   4. PHPUnit_TextUI_TestRunner->doRun() phar:///usr/local/bin/phpunit/phpunit/TextUI/Command.php:186
00:02:56.745 PHP   5. PHPUnit_Framework_TestSuite->run() phar:///usr/local/bin/phpunit/phpunit/TextUI/TestRunner.php:423
00:02:56.745 PHP   6. PHPUnit_Framework_TestSuite->run() phar:///usr/local/bin/phpunit/phpunit/Framework/TestSuite.php:751
00:02:56.745 PHP   7. PHPUnit_Framework_TestCase->run() phar:///usr/local/bin/phpunit/phpunit/Framework/TestSuite.php:751
00:02:56.748 PHP   8. PHPUnit_Framework_TestResult->run() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:722
00:02:56.751 PHP   9. PHP_Invoker->invoke() phar:///usr/local/bin/phpunit/phpunit/Framework/TestResult.php:641
00:02:56.751 PHP  10. call_user_func_array:{phar:///usr/local/bin/phpunit/php-invoker/Invoker.php:93}() phar:///usr/local/bin/phpunit/php-invoker/Invoker.php:93
00:02:56.752 PHP  11. PHPUnit_Framework_TestCase->runBare() phar:///usr/local/bin/phpunit/php-invoker/Invoker.php:93
00:02:56.752 PHP  12. PHPUnit_Framework_TestCase->runTest() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:766
00:02:56.752 PHP  13. ReflectionMethod->invokeArgs() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:881
00:02:56.752 PHP  14. Test_LocalAddressBook->testSearchId() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:881
00:02:56.752 PHP  15. OC\Contacts\LocalAddressBook->search() /var/jenkins/workspace/pull-request-analyser-ng-simple@3/label/SLAVE/tests/lib/contacts/localadressbook.php:41
```

**Comment by MorrisJobke** [comment]
@LEDfan Have a look at the failures ;)

**Comment by LEDfan** [comment]
I'm going to close this. The issue I had is solved by first searching for all contacts and then manually get only the one you need. (which makes it faster too).

**Pull request closed without merging**